okio/src/jvmMain/kotlin/okio/RealBufferedSource.kt

```diff
@@ -174,6 +175,24 @@ internal actual class RealBufferedSource actual constructor(
       override fun close() = this@RealBufferedSource.close()
 
       override fun toString() = "${this@RealBufferedSource}.inputStream()"
+
+      override fun transferTo(out: OutputStream): Long {
+        if (closed) throw IOException("closed")
+        var count = 0L
+        while (true) {
+          if (buffer.size == 0L) {
+            val read = source.read(buffer, Segment.SIZE.toLong())
+            if (read == -1L) break
+          }
+          count = try {
+            Math.addExact(count, buffer.size)
+          } catch (ignored: ArithmeticException) {
+            Long.MAX_VALUE
+          }
```

Comment:
I'm assuming the implementation of `addExact` is using a conditional to check for overflow and then throw an exception. So rather than run a conditional, throw, and catch, why don't we just do the conditional ourselves. Plus we know that neither value will ever be negative and can simply check for wrap-around.
```suggestion
          count += buffer.size
          if (count < 0) {
            count = Long.MAX_VALUE
          }
```

Reply:
LGTM, I just following some other implementations.